code/modules/mob/living/ventcrawling.dm

```diff
@@ -92,22 +92,22 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, typecacheof(list(
 	if(!totalMembers.len)
 		return
 
-	if(client)
-		for(var/X in totalMembers)
-			var/obj/machinery/atmospherics/A = X //all elements in totalMembers are necessarily of this type.
-			if(in_view_range(client.mob, A))
-				if(!A.pipe_vision_img)
-					A.pipe_vision_img = image(A, A.loc, dir = A.dir)
-					A.pipe_vision_img.plane = ABOVE_HUD_PLANE
-				client.images += A.pipe_vision_img
-				pipes_shown += A.pipe_vision_img
+	if(client) //MONKESTATION EDIT: Refactors Ventcawling to be less laggy
+		for(var/obj/machinery/atmospherics/member in range(8,get_turf(src)))
+			if(!istype(member))
+				continue
+			if(member in totalMembers)
+				if(!member.pipe_vision_img)
```

Comment:
`thing in list` loops over the list. The cost is based on `thing`'s position in the list. Total_members gets very large, and loop overhead is a thing. There is a non-zero chance this PR *increases* the cost.

Reply:
Baller as hell I'll close the PR and look into it when I have more time